lib/auth/recordingmetadata/recordingmetadatav1/service.go

```diff
@@ -0,0 +1,144 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package recordingmetadatav1
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/gravitational/teleport"
+	pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/recordingmetadata/v1"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/teleport/lib/player"
+	"github.com/gravitational/teleport/lib/session"
+)
+
+// Service implements the RecordingMetadataServiceServer interface, providing methods to retrieve session recording
+// metadata and thumbnails.
+type Service struct {
+	pb.UnimplementedRecordingMetadataServiceServer
+
+	authorizer    Authorizer
+	streamer      player.Streamer
+	uploadHandler events.UploadHandler
+	logger        *slog.Logger
+}
+
+// Authorizer is an interface that defines the method for authorizing access to session recordings.
+type Authorizer interface {
+	// Authorize checks if the user has permission to access the session recording, called with the session ID.
+	Authorize(context.Context, string) error
+}
+
+// ServiceConfig holds the configuration for the recording metadata service.
+type ServiceConfig struct {
+	// Authorizer is used to check if the user has permission to access the session recording.
+	Authorizer Authorizer
+	// Streamer is used to stream session recordings.
+	Streamer player.Streamer
+	// UploadHandler is used to handle uploads and downloads of session recording metadata and thumbnails.
+	UploadHandler events.UploadHandler
+}
+
+var _ pb.RecordingMetadataServiceServer = (*Service)(nil)
```

Comment:
we don't gain anything from this. `pb.UnimplementedRecordingMetadataServiceServer` already satisfies the server interface even if your methods are incorrectly named the interface is always satisfied by the unimplemented server.